backends/platform/ios7/ios7_osys_main.cpp

```diff
@@ -158,8 +158,10 @@ bool OSystem_iOS7::hasFeature(Feature f) {
 	switch (f) {
 	case kFeatureCursorPalette:
 	case kFeatureFilteringMode:
+#if TARGET_OS_IOS
 	case kFeatureVirtualKeyboard:
 	case kFeatureClipboardSupport:
+#endif
```

Comment:
I am still unsure about including `kFeatureVirtualKeyboard` in the #if here. I understood that there is no accessory view on tvOS (the bar we had on iOS to provide Fn keys, Esc key, and arrow keys), but I though that there is still the standard onscreen keyboard? In that case it should return true for the virtual keyboard feature.

And in any case this is inconsistent with `setFeatureState` and `getFeatureState` below that still do something on tvOS for the `kFeatureVirtualKeyboard` case. So either it should be excluded on tvOS in all three functions or included in all three functions.

Reply:
No, there is no onscreen keyboard shown in tvOS unfortunately. 
I have to either implement use of the standard input views in tvOS which are all full screen and rather clumsy. 

https://developer.apple.com/library/archive/documentation/General/Conceptual/AppleTV_PG/CreatingaGreatTextInputExperience.html

Or I implement a custom keyboard. I think the latter one would be preferred but requires some coding.

Anyway, the virtual keyboard is not supported and if a user wants to enter text a Bluetooth keyboard has to be paired with the Apple TV. But shame on me that I missed to  change `setFeatureState` and `getFeatureState`. I will push an update. Thanks for taking your time.

Reply:
Actually, thinking about it, this `kFeatureVirtualKeyboard` might be needed on tvOS as well (in all three functions). On iOS, this not only controls the onscreen keyboard (if no external keyboard is attached), but also bluetooth keyboards. Text input is not always on for those, and needs to be activated using this `kFeatureVirtualKeyboard` feature. I suspect this is the same on tvOS since they share so much code.

Reply:
Ok. I will test with a Bluetooth keyboard during the weekend and see if that is the case. 👍

Reply:
I think I should give some time to make the virtual keyboard work also for tvOS. It should, according to the documentation, pop up when becoming the first responder. But I guess something more is needed. So I will look into it.